core/src/main/scala/cats/data/OptionT.scala

```diff
@@ -87,55 +107,212 @@ final case class OptionT[F[_], A](value: F[Option[A]]) {
   def map[B](f: A => B)(implicit F: Functor[F]): OptionT[F, B] =
     OptionT(F.map(value)(_.map(f)))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT.some[List](99)
+   *  scala> optionT.imap[Char](_.toChar)(_.toInt)
+   *  res0: OptionT[List, Char] = OptionT(List(Some(c)))
+   * }}}
+   */
   def imap[B](f: A => B)(g: B => A)(implicit F: Invariant[F]): OptionT[F, B] =
     OptionT {
       F.imap(value)(_.map(f))(_.map(g))
     }
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> type ToString[A] = Function1[A, String]
+   *  scala> val optionIntToString: ToString[Option[Int]] = oi => oi.map(_.toString + "!").getOrElse("default")
+   *  scala> val optionT: OptionT[ToString, Int] = OptionT[ToString, Int](optionIntToString)
+   *  scala> optionT.contramap[Double](_.toInt).value(Some(5.4321))
+   *  res0: String = 5!
+   * }}}
+   */
   def contramap[B](f: B => A)(implicit F: Contravariant[F]): OptionT[F, B] =
     OptionT {
       F.contramap(value)(_.map(f))
     }
 
   /**
    * Modify the context `F` using transformation `f`.
+   * 
+   * Example:
+   * {{{
+   *  scala> import cats.~>
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   * 
+   *  scala> val optionToList: Option ~> List = new ~>[Option, List] { override def apply[A](o: Option[A]): List[A] = o.toList }
+   *  scala> val optionT: OptionT[Option, Int] = OptionT.some(42)
+   *  scala> optionT.mapK[List](optionToList)
+   *  res0: OptionT[List, Int] = OptionT(List(Some(42)))
+   * }}}
    */
   def mapK[G[_]](f: F ~> G): OptionT[G, A] = OptionT[G, A](f(value))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT[List, Int](List(Some(3), Some(5)))
+   *  scala> optionT.semiflatMap(x => List(x, x * 6))
+   *  res0: OptionT[List, Int] = OptionT(List(Some(3), Some(18), Some(5), Some(30)))
+   * }}}
+   */
   def semiflatMap[B](f: A => F[B])(implicit F: Monad[F]): OptionT[F, B] =
     flatMap(a => OptionT.liftF(f(a)))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.data.OptionT
+   *  scala> import scala.util.Try
+   *
+   *  scala> // prints "5"
+   *  scala> val optionT: OptionT[Try, Int] = OptionT[Try, Int](Try(Some(5)))
+   *  scala> optionT.semiflatTap(x => Try(println(x))).value
+   *  res0: Try[Option[Int]] = Success(Some(5))
+   * }}}
+   */
   def semiflatTap[B](f: A => F[B])(implicit F: Monad[F]): OptionT[F, A] =
     semiflatMap(a => F.as(f(a), a))
 
+  /**
+   * Example:
+   * {{{
+   *  scala> import cats.implicits._
+   *  scala> import cats.data.OptionT
+   *
+   *  scala> val optionT: OptionT[List, Int] = OptionT[List, Int](List(Some(2), Some(3), Some(4)))
+   *  scala> optionT.mapFilter(x => Option.when(x % 2 == 0)(x))
```

Comment:
CI might not like `Option.when` on Scala 2.12. I guess we'll find out :)

Reply:
Correct, I changed to `Option(...).filter(...)`.
I assume with `~+coreJVM/testOnly **OptionTDoctest` (added the `+`) I should catch those problems, right?

Reply:
Possibly. I don't know how `~` and `+` interact in sbt.